pkg/asset/installconfig/azure/validation.go

```diff
@@ -188,11 +192,24 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 		if ultraSSDCapability == "" {
 			ultraSSDCapability = defaultUltraSSDCapability
 		}
+
+		capabilities, err = client.GetVMCapabilities(context.TODO(), instanceType, ic.Azure.Region)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
+		}
+
 		if vmNetworkingType == "" {
-			vmNetworkingType = defaultVMNetworkingType
+			val, ok := capabilities[string(azure.AcceleratedNetworkingEnabled)]
+			if ok {
+				if strings.EqualFold(val, "True") {
+					vmNetworkingType = string(azure.VMnetworkingTypeAccelerated)
+				} else {
+					vmNetworkingType = string(azure.VMNetworkingTypeBasic)
+				}
+			}
```

Comment:
I'm not aware of a need to set the default here when validating. To me the approach that makes sense is:
- if a user sets the value, validate it is possible
- if not, skip that validation, set the default value that works in the machines

Reply:
I agree, I will not set the default. The only thing that is redundant is when the value is not set we are setting it to a default value based upon the capabilities and then validating the default that we just set.

Reply:
> The only thing that is redundant is when the value is not set we are setting it to a default value based upon the capabilities and then validating the default that we just set.

Maybe there's a disconnect or I'm missing something. In what I'm suggesting I don't think we would be: " validating the default that we just set."

My suggestion is that we:
- validate the user provider install config (if accelerated networking is set, validate that too, if not skip it)
- set a default, if needed, for the machinesets

I don't think those machinesets would be validated.